CHANGELOG.md

```diff
@@ -1,5 +1,14 @@
 ## Unreleased
 
+## v0.21.2
+IMPROVEMENTS:
+* Update dependencies:
+  * `golang.org/x/net` v0.35.0 -> v0.36.0: [GH-245](https://github.com/hashicorp/vault-plugin-secrets-gcp/pull/245)
+  * Update Go version to 1.23.6
```

Comment:
Technically the go build version was already updated to 1.23.6 in a previous release via the .go-version file. The go.mod file just specifies the minimum version required to build the plugin. I don't think this is really worth changing though.

Reply:
Ah gotcha, thanks for the detail! I added the change because I was getting a Security Scan issue locally, but good to know that it shouldn't affect CI builds 👍🏼